src/compiler/crystal/codegen/debug.cr

```diff
@@ -3,14 +3,15 @@ require "./codegen"
 module Crystal
   class CodeGenVisitor
     CRYSTAL_LANG_DEBUG_IDENTIFIER = 0x8002_u32
+    CPP_LANG_DEBUG_IDENTIFIER     = 0x0004_u32
 
     @current_debug_location : Location?
 
     def di_builder(llvm_module = @llvm_mod || @main_mod)
       di_builders = @di_builders ||= {} of LLVM::Module => LLVM::DIBuilder
       di_builders[llvm_module] ||= LLVM::DIBuilder.new(llvm_module).tap do |di_builder|
         file, dir = file_and_dir(llvm_module.name == "" ? "main" : llvm_module.name)
-        di_builder.create_compile_unit(CRYSTAL_LANG_DEBUG_IDENTIFIER, file, dir, "Crystal", 0, "", 0_u32)
+        di_builder.create_compile_unit(CPP_LANG_DEBUG_IDENTIFIER, file, dir, "Crystal", false, "", 0_u32)
```

Comment:
Could you add a comment explaining why we are using CPP? I'm almost sure it's because if we do that we get a lot of goodies, but a comment will help us know this in the future.

Reply:
We have to use it because it has builtin type system support that we can use for now for free.
I still have to implement Crystal type system and language support to do it Crystal way (mostly path evaluators) but initial small fix allows us to use C++ type system and see the values right away.

Reply:
Great, thank you for the explanation! You can add it as a comment in the code, or we can merge this PR and later I'll send another PR which adds the comments. As you wish!

Reply:
Crystal isn't C++, we don't implement anything remotely similar to C++, and *really* shouldn't have to specify a foreign language for a debugger to pick up variables, types and displaying them. It should only be useful for some language niceties (e.g. demangling function names).

Declaring custom types & structures as DWARF (which we currently don't) should be enough. GDB, for example, doesn't have any problem with a custom language identifier.

Reply:
> Great, thank you for the explanation! You can add it as a comment in the code, or we can merge this PR and later I'll send another PR which adds the comments. As you wish!

You can merge it and I will add comment later when I will create a new PR with some extra features for proper showing some of the variables.

 I also will try to look at GDB code and check why it is failing to stop on breakpoints on macOS.